modules/openapi-generator/src/main/resources/dart-dio/pubspec.mustache

```diff
@@ -4,14 +4,17 @@ description: {{pubDescription}}
 environment:
     sdk: '>=2.7.0 <3.0.0'
 dependencies:
-    dio: ^3.0.9
+    dio: '^3.0.9'
     built_value: '>=7.1.0 <8.0.0'
     built_collection: '>=4.3.2 <5.0.0'
 {{#timeMachine}}
-    time_machine: ^0.9.12
+    time_machine: '^0.9.12'
 {{/timeMachine}}
 dev_dependencies:
     built_value_generator: '>=7.1.0 <8.0.0'
-    build_runner: ^1.7.1
+    build_runner: any
     test: '>=1.3.0 <1.16.0'
+dependency_overrides:
+    # Needed to make built_value work correctly with Dart 2.12
```

Comment:
not sure we want to do that... if we start generating stuff like this it probably mean some deps are not up to date so we should wait until those deps are fixed. You can do that on your own project, but not push that for everyone. Let's see what opinion have other reviewers. I would probably push more to have a null safety generation once dio is ready

Reply:
The problem is we have Dart SDK 2.12 in CI and there are not gonna be any new dependency versions for this generator, ever. There is no migration path for the dart-dio generator to 2.12/NNBD. It simply needs to be replaced by the dart-dio-next generator and this one should be deprecated.
We could leave it as is but then we need Dart 2.10 in the CI.

Reply:
```
Your current `analyzer` version may not fully support your current SDK version.

Please try upgrading to the latest `analyzer` by running `flutter packages upgrade`.

Analyzer language version: 2.10.0
SDK language version: 2.12.0

If you are getting this message and have the latest analyzer please file
an issue at https://github.com/dart-lang/sdk/issues/new with the title
"No published analyzer available for language version 2.12.0".
Please search the issue tracker first and thumbs up and/or subscribe to
existing issues if present to avoid duplicates.
```

Reply:
I reverted the changes with the analyzer. I can't get it to build locally with Dart 2.12.0 or 2.12.2 but in CI it only seems to be a warning.